### DISCUSSION TITLE
Time to finally drop support for powershell 3?

### DISCUSSION THREAD

**Discussion by nohwnd** [pester/Pester#2431] [General]
I am not wrong, most of the versions of PowerShell are now officially out of support, except for Windows PowerShell 5.1, PowerShell 7.2, 7.3 (ends next month) and 7.4. The time has finally come to stop building on PowerShell 3, 4, 6.2.

**Comment by nohwnd** [comment]
@fflaten I am basing this on this table and the info I can piece out from the msft docs: 

https://endoflife.date/powershell

https://learn.microsoft.com/en-us/powershell/scripting/install/powershell-support-lifecycle?view=powershell-7.4

https://learn.microsoft.com/en-us/powershell/scripting/windows-powershell/wmf-overview?view=powershell-7.4

I also think there was some issue to do this, but I cannot find it.

**Reply by nohwnd** [reply]
https://github.com/pester/Pester/pull/2432

**Comment by fflaten** [comment]
I'm all for it. Related #2324 👍

**Reply by nohwnd** [reply]
That is the issue I was looking for!

**Comment by kilasuit** [comment]
PowerShell is a part of Windows Server & the ESU's for Server 2012 (which shipped v3) are still in support that Ends in 2 years and 6 months
(13 Oct 2026) so whilst very unlikely there may be some security patches for it still

**Reply by fflaten** [reply]
ESU is expensive custom support (3 years included in Azure). The OS itself is EOL, so at this point users should be able to pin the Pester to 5.5.0 imo. Do you agree?

**Reply by nohwnd** [reply]
I do agree. This is a free to use tool. If someone has the needs I am open to do paid support.
